Add packaging functionality

gaborod16 opened the pull request:
Integration of packaging capabilities for portability

nightah committed:
Update grafana to v8.1.1

nightah committed:
Update prometheus to v2.29.0

nightah committed:
Update grafana to v8.1.2

nightah committed:
Update alertmanager to v0.23.0

Signed-off-by: Amir Zarrinkafsh <nightah@me.com>

dlh committed:
fix(cadvisor): mount /var/run as read-only (#191)

This is in-line with the example from the cadvisor README:

https://github.com/google/cadvisor/blob/master/README.md#quick-start-running-cadvisor-in-a-docker-container

nightah committed:
fix(cadvisor): add /dev/kmsg device for oom events

Also include priviledged flag per cadvisor recommendations.

Closes #215.
Closes #232.

Signed-off-by: Amir Zarrinkafsh <nightah@me.com>

nightah committed:
build(deps): update dependency prometheus to v2.29.2

Signed-off-by: Amir Zarrinkafsh <nightah@me.com>

nightah committed:
build(deps): update dependency grafana to v8.1.3

nightah committed:
Update prometheus to v2.30.0

sakkiii committed:
Update grafana to v8.1.4 (#240)

nightah committed:
Update grafana to v8.1.5

nightah committed:
Update prometheus to v2.30.1

nightah committed:
Update cadvisor to v0.41.0

nightah committed:
build(deps): update cadvisor to v0.41.0

nightah committed:
Update prometheus to v2.30.3

nightah committed:
revert: Update cadvisor to v0.41.0

This reverts commit 0058c9090f6a6435addb828b90bcbe9f954341a2.

nightah committed:
build(deps): update dependency grafana to v8.2.0

nightah committed:
build(deps): update grafana to v8.2.1

nightah committed:
build(deps): update pushgateway to v1.4.2

nightah committed:
build(deps): update cadvisor to v0.42.0

sakkiii committed:
build(deps): update grafana to v8.2.2 (#244)

nightah committed:
build(deps): update prometheus to v2.31.1

nightah committed:
build(deps): update cadvisor to v0.43.0

nightah committed:
build(deps): update grafana to v8.2.5

howiezhao committed:
fix: grafana provisioning file path in readme (#245)

nightah committed:
build(deps): update node-exporter to v1.3.0

Commit:
WIP: packaging of the monitoring system

Commit:
Extend README.md with the packaging instructions

Commit:
Update .gitignore

Commit:
Add packaging folder and Dockerfile

Commit:
Update .gitignore

Commit:
- Add Makefile for easy usage of the packaging
- Add example.env to be used as base

Commit:
Update .gitignore

Commit:
Re-organise the Makefile and allow the saving of container dependencies

Commit:
Add docker-compose.packaging.yml for registry/packaged installations

Commit:
Extend the packaging dockerfile to account for the packaging folder and the container dependencies

Commit:
Merge branch 'add_packaging_functionality' of github.com:ondewo/ondewo-monitoring into add_packaging_functionality

Commit:
Fix permissions issue

Commit:
Extend the README.md and renaming of file for better understandability

gaborod16 committed:
Merge branch 'stefanprodan:master' into add_packaging_functionality

Commit:
pump up versions

Commit:
pump up versions

Commit:
Merge branch 'add_packaging_functionality' of github.com:ondewo/ondewo-monitoring into add_packaging_functionality

ondewo-jenkins commented on docker-compose.published_dependencies.yml:
```diff
@@ -0,0 +1,135 @@
+    container_name: pushgateway
+    restart: unless-stopped
+    expose:
+      - 9091
```
Add port into environment file

teddius replied:
Add port into environment file

teddius commented on docker-compose.published_dependencies.yml:
```diff
@@ -0,0 +1,135 @@
+      - '--web.enable-lifecycle'
+    restart: unless-stopped
+    expose:
+      - 9090
```
Add port into environment file

teddius commented on docker-compose.published_dependencies.yml:
```diff
@@ -0,0 +1,135 @@
+      - '--storage.path=/alertmanager'
+    restart: unless-stopped
+    expose:
+      - 9093
```
Add port into environment file

teddius commented on docker-compose.published_dependencies.yml:
```diff
@@ -0,0 +1,135 @@
+      - '--collector.filesystem.mount-points-exclude=^/(sys|proc|dev|host|etc)($$|/)'
+    restart: unless-stopped
+    expose:
+      - 9100
```
Add port into environment file

teddius commented on docker-compose.published_dependencies.yml:
```diff
@@ -0,0 +1,135 @@
+      #- /cgroup:/cgroup:ro #doesn't work on MacOS only for Linux
+    restart: unless-stopped
+    expose:
+      - 8080
```
Add port into environment file

teddius commented on docker-compose.published_dependencies.yml:
```diff
@@ -0,0 +1,135 @@
+      - GF_USERS_ALLOW_SIGN_UP=false
+    restart: unless-stopped
+    expose:
+      - 3000
```
Add port into environment file

teddius commented on docker-compose.published_dependencies.yml:
```diff
@@ -0,0 +1,135 @@
+  caddy:
+    image: ${PROD_CADDY_IMAGE_NAME}
+    container_name: caddy
+    ports:
```
Add ports into environment file

teddius commented on packaging/example.env:
```diff
@@ -0,0 +1,38 @@
+# Packaging
```
Add ports into this environment file so it is possible to configure the ports from the environment file

Commit:
- Update Docker compose version
 - Add configurable ports to the .env file and docker-compose files
 - Update make recipe to use the .env file on with the docker-compose call

Commit:
- Add recipe to stop the containers (Note: the .env files used may differ, so it is easier than using docker-compose directly)
 - Allow the dynamic prefix naming for the docker containers

Commit:
Simplify naming given as feedback from @MohamedYaman

MohamedYaman reviewed (approved):
Good job Gabe!
